elide-datastore/elide-datastore-hibernate5/src/main/java/com/yahoo/elide/datastores/hibernate5/HibernateTransaction.java

```diff
@@ -207,6 +252,71 @@ public void commit() {
     }
 
     @Override
+    public <T> Collection filterSortOrPaginateCollection(final Collection collection, final Class<T> entityClass,
+                                                         final EntityDictionary dictionary,
+                                                         final Optional<Set<Predicate>> filters,
+                                                         final Optional<Sorting> sorting,
+                                                         final Optional<Pagination> pagination) {
+        if (((collection instanceof AbstractPersistentCollection))
+                && (filters.isPresent() || sorting.isPresent() || pagination.isPresent())) {
+
+            String filterString = "";
+
+            // apply filtering - eg where clause's
+            if (filters.isPresent()) {
+                filterString += hqlFilterOperation.applyAll(filters.get());
+            }
+
+            // add sorting into HQL string query generation
+            if (sorting.isPresent() && !sorting.get().isEmpty()) {
+
+                final Map<String, Sorting.SortOrder> validSortingRules = sorting.get().getValidSortingRules(
+                        entityClass, dictionary
+                );
+                String additionalHQL = "";
+                if (!validSortingRules.isEmpty()) {
```

Comment:
sure. yeah, it is same as ```for (val : emptyCollection)```. However, since this is an optional for sorting, I didn't want to do additional work and create unnecessary garbage in the case of the default sorting internal map.